<!-- vim: set filetype=xml : -->
<launch>
  <arg name="config_file" default=""/>
  <arg name="results_dir" default="/home/dev/ros_ws/results"/>
  <arg name="run_name" default="rko_lio_online_run"/>
  <arg name="publish_deskewed_cloud" default="true"/>
  <arg name="rviz" default="false"/>
  <arg name="rviz_config_file" default="ros1.rviz"/>

  <node pkg="rko_lio" type="online_node" name="rko_lio_online_node" output="screen" required="true">
    <!-- Load parameters from a YAML file -->
    <rosparam file="$(find rko_lio)/config/$(arg config_file)" command="load" />
    <param name="results_dir" value="$(arg results_dir)" />
    <param name="run_name" value="$(arg run_name)" />
    <param name="publish_deskewed_cloud" value="$(arg publish_deskewed_cloud)" />
  </node>
    
  <group if="$(arg rviz)">
    <node pkg="rviz" type="rviz" name="rviz" 
          args="-d $(find rko_lio)/config/$(arg rviz_config_file)" output="screen"/>
  </group>
</launch>


